File: core/src/com/questmark/entity/components/PreviousPositionComponent.java
package com.questmark.entity.components;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.math.Vector2;

/**
 * Represents the position of an entity in (x,y) coordinates on the map before a delta time velocity update.
 * Used for entity collision detection.
 * Can be defined as either a {@link Vector2} or two floats x, y.
 *
 * @author Ming Li
 */
public final class PreviousPositionComponent implements Component {

    public Vector2 p;

    public PreviousPositionComponent(float x, float y) {
        p = new Vector2(x, y);
    }

    public PreviousPositionComponent(Vector2 pos) {
        this(pos.x, pos.y);
    }

}

File: core/src/com/questmark/entity/components/PreviousPositionComponent.kt
package com.questmark.entity.components

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.math.Vector2

/**
 * Represents the position of an entity in (x,y) coordinates on the map before a delta time velocity update.
 * Used for entity collision detection.
 * Can be defined as either a [Vector2] or two floats x, y.
 *
 * @author Ming Li
 */
class PreviousPositionComponent(x: Float, y: Float) : Component {

    var p: Vector2 = Vector2(x, y)

    constructor(pos: Vector2) : this(pos.x, pos.y)

}
